### PR TITLE
provider_test.go upgraded to Sepolia

### PR DESCRIPTION
**Pull request by thiagodeev**
Started upgrading the RPC tests to Sepolia network instead of Goerli.
This PR upgrades the provider_test.go tests to Sepolia.
Partially resolves the issue https://github.com/NethermindEth/starknet.go/issues/551.

### CODE REVIEW

**Comment by thiagodeev** [comment]
The fixed test is breaking with the 'devnet' param, the devnet is returning 'goerli' as the chainId. I didn't test with devnet, my bad.
I will switch the current legacy devnet container to the new recommended devnet-rs container.

**Comment by rianhughes** [comment]
Hey, the PR looks good, you just need to fix the TestCall that is failing on devnet

File: devnet/devnet.go [hunk 1]
```diff
@@ -107,8 +112,9 @@ func (devnet *DevNet) IsAlive() bool {
 }
 
 type MintResponse struct {
-	NewBalance *big.Int `json:"new_balance"`
-	Unit       string   `json:"unit"`
+	NewBalance      string `json:"new_balance"`
```
Comment (thiagodeev) [inline devnet/devnet.go:115]: The new devnet version returns a string instead a number, and a new field. The compiler was unable to convert to bigInt

File: devnet/devnet_test.go [hunk 2]
```diff
@@ -44,26 +48,6 @@ func TestDevnet_Accounts(t *testing.T) {
 	}
 }
 
-// TestDevnet_FeeToken tests the FeeToken function of the Devnet struct.
```
Comment (rianhughes) [inline devnet/devnet_test.go:47]: Why was this removed? I think we can add it back in without breaking tests

Comment (thiagodeev) [reply devnet/devnet_test.go:47]: I removed this tests because the[ new devnet doesn't have the "/fee_token" endpoint anymore](https://github.com/0xSpaceShard/starknet-devnet-rs/blob/main/crates/starknet-devnet-server/src/server.rs#L34). That being said, I believe the "Mint" and "TestCall" tests are enough to prove the token existing, address correctness, and functionalities.

File: rpc/call_test.go [hunk 3]
```diff
@@ -38,12 +38,12 @@ func TestCall(t *testing.T) {
 			{
 				FunctionCall: FunctionCall{
 					// ContractAddress of predeployed devnet Feetoken
-					ContractAddress:    utils.TestHexToFelt(t, "0x49d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7"),
-					EntryPointSelector: utils.GetSelectorFromNameFelt("name"),
+					ContractAddress:    utils.TestHexToFelt(t, DevNetETHAddress),
```
Comment (thiagodeev) [inline rpc/call_test.go:41]: Using an existing variable instead of hardcoding the address

File: rpc/call_test.go [hunk 4]
```diff
@@ -38,12 +38,12 @@ func TestCall(t *testing.T) {
...
+					EntryPointSelector: utils.GetSelectorFromNameFelt("decimals"),
```
Comment (thiagodeev) [inline rpc/call_test.go:42]: I had some issues related to felt.Felt comparison using the "name" method. Using "decimals" was simpler to compare

**Review by rianhughes — APPROVED** [review]
